### PR TITLE
Change appearance of the output help messages

### PR DESCRIPTION
**Pull request by desa85**

### CODE REVIEW

File: src/help.js [hunk 1]
```diff
@@ -99,14 +103,90 @@ module.exports = (robot) => {
...
+      title: group,
+      text: commandText.join('\n')
+    })
+  }
+  for (let group in groupCommand) {
+    if (group !== 'Other commands') {
+      makeRichMessage(group)
+    }
+  }
+  makeRichMessage('Other commands')
```
Comment (polina-popova) [inline src/help.js]: You should check if this key exist.

**Review by polina-popova — COMMENTED** [review]
Requested changes are:
1. write docstrings to functions,
2. doing this try to manage more relevant names for variables and functions 
3. and fix its English translation,
4. follow camel case style in functions names,
5. translate errors text,
6. update README.

**Review by polina-popova — COMMENTED** [review]
Still needed:
1. follow camel case style in functions names,
2. translate errors text,
3. manage more relevant names for variables and functions
3. and let's split WalkInGroups function to several independent functions (because you don't use the enclosing namespase).

File: src/help.js [hunk 2]
```diff
@@ -99,14 +103,98 @@ module.exports = (robot) => {
...
-  return str.replace(rights, '`privileged: admins only`')
+  if (!str.match(/^(begin|end)/i)) {
+    str = '*' + str
+    str = str.replace(/ - /i, ' *- ')
+  }
+  return str
+}
+
+/**
+ * parses an array with commands and collects an object with keys called the group name.
```
Comment (polina-popova) [inline src/help.js]: Start sentence from big letter.
`Parses the array with commands and write groups names and relevant commands to the object.`

File: src/help.js [hunk 3]
```diff
@@ -99,14 +103,98 @@ module.exports = (robot) => {
...
+ *
+ * @param {Robot} robot - Hubot instance.
+ * @param {array} commands - Array of commands.
+ * @param {string} markerName - The name of the marker that will initialize a group of commands.
```
Comment (polina-popova) [inline src/help.js]: `The name of the marker that mark a start of commands group.`

File: src/help.js [hunk 4]
```diff
@@ -99,14 +103,98 @@ module.exports = (robot) => {
...
+ *
+ * @returns {object}
+ */
+var groupingByCommands = function groupingByCommands (robot, commands, markerName) {
```
Comment (polina-popova) [inline src/help.js]: `var groupByMarkerGroupName`

File: src/help.js [hunk 5]
```diff
@@ -99,14 +103,98 @@ module.exports = (robot) => {
...
+  let err
+  let inBegin = false
+  let inGroup = false
+  let groupCommand = {}
```
Comment (polina-popova) [inline src/help.js]: `let commandsObject`

File: src/help.js [hunk 6]
```diff
@@ -99,14 +103,98 @@ module.exports = (robot) => {
...
+  let marker
+  let markerGroup
```
Comment (polina-popova) [inline src/help.js]: `let markerGroupName`

File: src/help.js [hunk 7]
```diff
@@ -99,14 +103,98 @@ module.exports = (robot) => {
...
+ * @param {string} markerName - The name of the marker that will initialize a group of commands.
+ *
+ * @returns {object}
+ */
+var groupingByCommands = function groupingByCommands (robot, commands, markerName) {
+  let err
+  let inBegin = false
+  let inGroup = false
+  let groupCommand = {}
+  let marker
```
Comment (polina-popova) [inline src/help.js:128]: `let markerLine`

File: src/help.js [hunk 8]
```diff
@@ -99,14 +103,98 @@ module.exports = (robot) => {
...
+  let pushInGroup = (command, markerGroup) => {
+    if (!groupCommand[markerGroup]) {
+      groupCommand[markerGroup] = []
+    }
+    groupCommand[markerGroup].push(command)
+  }
+
+  commands.map((command, index) => {
+    marker = command.match(markerReg)
+    err = `${markerGroup ? `in the script "${markerGroup}" is` : `in some script the`} closing marker was not added`
```
Comment (polina-popova) [inline src/help.js]: ```${markerGroup ? `In the script "${markerGroup}" ` : `In some script the`} closing marker was not found. ```

File: src/help.js [hunk 9]
```diff
@@ -99,14 +103,98 @@ module.exports = (robot) => {
...
+    if (marker) {
+      if ((inGroup && (marker[1] + marker[2]) === 'begingroup')) {
+        throw (routines.rave(robot, err))
+      }
+      if (!inBegin && marker[0] === `end ${markerName}`) {
+        markerGroup = !inBegin ? false : markerGroup
+        err = `${markerGroup ? `in the script "${markerGroup}"` : `in some script the`} opening marker was not added`
```
Comment (polina-popova) [inline src/help.js]: ```${markerGroup ? `In the script "${markerGroup}"` : `In some script the`} opening marker was not found.```
Looks very similar to error text upper, let's unite them ti one variable.

File: src/help.js [hunk 10]
```diff
@@ -99,14 +103,98 @@ module.exports = (robot) => {
...
+        throw (routines.rave(robot, err))
+      }
+
+      inBegin = marker[1] === 'begin'
```
Comment (polina-popova) [inline src/help.js]: `inOpeningMarker`

File: src/help.js [hunk 11]
```diff
@@ -99,14 +103,98 @@ module.exports = (robot) => {
...
+      inGroup = inBegin && marker[2] === markerName
```
Comment (polina-popova) [inline src/help.js]: `inMarkerGroup`

File: src/help.js [hunk 12]
```diff
@@ -99,14 +103,98 @@ module.exports = (robot) => {
...
+      } else {
+        pushInGroup(command, 'Other commands')
+      }
+    }
+  })
+  return groupCommand
+}
+
+/**
+ * Constructs a rich messages.
```
Comment (polina-popova) [inline src/help.js]: `* Constructs the array of rich messages.`
And you need to describe the `groupCommand` argument.

File: src/help.js [hunk 13]
```diff
@@ -56,10 +56,13 @@ const helpContents = (name, commands) => `\
...
 </html>\
 `
+const routines = require('hubot-routines')
+
+let groupCommand
 
 module.exports = (robot) => {
   robot.respond(/help(?:\s+(.*))?$/i, (msg) => {
-    let cmds = getHelpCommands(robot)
+    let cmds = getHelpCommands(robot, msg.message.user.name.toString())
```
Comment (ya7on) [inline src/help.js]: `msg.message.user.name` is already string. `toString()` is no needed

File: src/help.js [hunk 14]
```diff
@@ -76,13 +79,15 @@ module.exports = (robot) => {
...
+      groupCommand = groupByMarkerGroupName(robot, emit.split('\n'), 'group')
+      return msg.send(makeRichMessage(groupCommand))
     }
   })
 
   if (process.env.HUBOT_HELP_DISABLE_HTTP == null) {
     return robot.router.get(`/${robot.name}/help`, (req, res) => {
-      let cmds = getHelpCommands(robot).map(cmd => cmd.replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;'))
+      let msg
+      let cmds = getHelpCommands(robot, msg.message.user.name.toString()).map(cmd => cmd.replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;'))
```
Comment (ya7on) [inline src/help.js]: `msg.message.user.name` is string too

File: src/help.js [hunk 15]
```diff
@@ -97,9 +102,109 @@ module.exports = (robot) => {
...
+var groupByMarkerGroupName = function groupByMarkerGroupName (robot, commands, markerName) {
+  let errStatus
+  let inOpeningMarker = false
+  let inMarkerGroup = false
+  let commandsObject = {}
+  let marker
+  let markerMatch
+  let markerRole
+  let markerFullName
+  let groupName
```
Comment (ya7on) [inline src/help.js]: Alphabetical order is required

File: README.md [hunk 16]
```diff
@@ -45,3 +45,4 @@ Sample Interaction
 hubot> hubot help - Displays all of the help commands that this bot knows about.
 hubot> hubot help <query> - Displays all help commands that match <query>.
 ```
+
```
Comment (polina-popova) [inline README.md]: Let's avoid this kind of changes - delete the blank string back.

File: src/help.js [hunk 17]
```diff
@@ -99,14 +103,108 @@ module.exports = (robot) => {
...
+        if (commands.length === index + 1) {
+          errStatus = `closing`
+        }
+        pushInGroup(command, markerGroupName)
+      } else {
+        pushInGroup(command, 'Other commands')
+      }
+    }
+    if (errStatus) {
+      throw (routines.rave(robot, `${markerGroupName ? `In the script "${markerGroupName}" ` : `In some script the`} ${errStatus} marker was not found.`))
```
Comment (polina-popova) [inline src/help.js]: Let's move `the` to the common part of the error message:
```${markerGroupName ? `In the script "${markerGroupName}" ` : `In some script`} the ${errStatus} marker was not found.```

File: src/help.js [hunk 18]
```diff
@@ -99,14 +103,108 @@ module.exports = (robot) => {
...
+    }
+  })
+  return commandsObject
+}
+
+/**
+ * Constructs the array of rich messages.
```
Comment (polina-popova) [inline src/help.js]: Choose one form of the verb in docstring ( with `s` at the end or without).

**Review by polina-popova — DISMISSED** [review]

**Review by eugulixes — APPROVED** [review]

**Pull request merged**